Druid/DRBytecodeBlockGenerator.class.st

```diff
@@ -0,0 +1,73 @@
+Class {
+	#name : 'DRBytecodeBlockGenerator',
+	#superclass : 'DRBytecodeGenerator',
+	#category : 'Druid-BytecodeGeneration',
+	#package : 'Druid',
+	#tag : 'BytecodeGeneration'
+}
+
+{ #category : 'compiling' }
+DRBytecodeBlockGenerator >> compileBlockFrom: aDRControlFlowGraph withScope: aScope [ 
+	
+	| ir compiledBlock|
+	
+	ir := builder ir.
+	1 halt.
+	"add vectorTempVar to the scope of the block"
+	"aDRControlFlowGraph allVectorTemporaryInstructions do: [ :var |
+		aScope addTemp: var.
+		 ]."
+	compiledBlock := ir compiledBlock: aScope.
+	
+	^ compiledBlock 
+]
+
+{ #category : 'code-generation' }
+DRBytecodeBlockGenerator >> compileBlockWithScope: aScope [
+
+	| ir compiledBlock|
+	
+	ir := builder ir.
+	compiledBlock := ir compiledBlock: aScope.
+	
+	^ compiledBlock 
+
+]
+
+{ #category : 'code-generation' }
+DRBytecodeBlockGenerator >> generateTargetASTFromIR: aDRControlFlowGraph [
+
+	| args temps vectorTemps temporariesList |
+	"Construct the tempVector"
+	args := aDRControlFlowGraph argumentNames.
+	temps := aDRControlFlowGraph temporaryVariableNames. 
+	vectorTemps := aDRControlFlowGraph vectorTemporaryVariableNames .
```

Comment:
Maybe this method requires a nice comment explaining the strategy for the temp vector?